version: "3.8"
services:
  alist:
    image: xhofe/alist:latest
    container_name: embygroup.alist
    restart: always
    ports:
      - "5244:5244"
    environment:
      - PUID=0
      - PGID=0
      - LMASK=022
    volumes:
      # docker exec -it embygroup.alist ./alist admin set password
      - /home/embygroup/alist:/opt/alist/data
    network_mode: bridge

  clouddrive2:
    image: cloudnas/clouddrive2
    container_name: embygroup.clouddrive2
    restart: unless-stopped
    ports:
      - "19798:19798"
    environment:
      - CLOUDDRIVE_HOME=/Config
    volumes:
      - /mnt/:/mnt:shared
      - /home/embygroup/clouddrive2:/Config
    devices:
      - /dev/fuse:/dev/fuse
    privileged: true
    network_mode: bridge

  embyserver:
    image: amilys/embyserver_arm64v8
    container_name: embygroup.embyserver
    restart: always
    ports:
      - "8096:8096"
    environment:
      - TZ=Asia/Shanghai
      - PUID=0
      - PGID=0
    volumes:
      - /home/embygroup/emby:/config
      - /mnt:/mnt:shared
    depends_on:
      - clouddrive2
      - alist
    network_mode: bridge

  auto_symlink:
    image: shenxianmq/auto_symlink:latest
    container_name: embygroup.auto_symlink
    restart: unless-stopped
    ports:
      - "8095:8095"
    environment:
      - TZ=Asia/Shanghai
    volumes:
      - /mnt/:/mnt:rslave
      - /home/embygroup/symlink:/app/config
    user: "0:0"
    network_mode: bridge
